src/vehicle_use.cpp

```diff
@@ -1485,7 +1485,14 @@ veh_interact_results vehicle::interact_with( const tripoint &pos, int interact_p
         if( fuel_left( "battery" ) < pseudo.ammo_required() ) {
             return false;
         }
-        auto qty = pseudo.ammo_capacity() - discharge_battery( pseudo.ammo_capacity() );
+        auto capacity = 0;
+        if( pseudo.magazine_default() != "null" ) {
```

Comment:
This treats the symptom (presumably, only one of many to come), not the cause. I suppose, `item::ammo_capacity()` is what needs fixing after #29651.

Reply:
At first I wanted to object to this, since this is a special case of an imaginary object wanting to store a concrete amount of power possibly in an imaginary magazine, but I realized that even still there should be a more consistent way of making that request to the `item` class than ad-hoc stuff like this scattered everywhere. I've got an idea for pursuing that.

Reply:
Added an argument to adjust the behavior of `item::ammo_capacity()`, and an overload with no argument that invokes the old behavior.